bob/Bob.cs

```diff
@@ -18,86 +18,76 @@ public static string Hey(string statement)
         response.Add(2, "Sure.");
         response.Add(3, "Fine. Be that way!");
 
+        if (ContainsNewLine(statement) == true)
+            return response[0].ToString();
+
+        if (IsQuestionMarkNotAtEndOfSentence(statement) == true)
+            return response[0].ToString();
+
         if (IsAllUpperCaseLettersOnly(statement) == true)
             return response[1].ToString();
 
         if (IsAllUpperCaseLettersAndSymbols(statement) == true)
             return response[1].ToString();
 
+        if (ContainsANumberBeforeAWordAndExclamationPoint(statement) == true)
+            return response[1].ToString();
+
         if (ContainsNumberAndQuestionMark(statement) == true)
             return response[2].ToString();
 
         if (IsLowerCaseQuestion(statement) == true)
             return response[2].ToString();
 
-        //if (HasDoubleWhiteSpaceAfterWord(statement) == true)
-        //    return response[2].ToString();
-
-        //if (IsStringEmpty(statement) == true)
-        //    return response[3].ToString();
+        if (ContainsThreeOrMoreSentences(statement) == true)
+            return response[2].ToString();
```

Comment:
This is starting to get confusing with the numbers. What exactly is the point of the dictionary?

Couldn't you just put the responses in properly named variables and return those variables? This seems like a bunch of overhead for virtually no gain.